src/components/fields/range-select/range-select.tsx

```diff
@@ -85,10 +85,19 @@ export const RangeSelect = (props: IGenericFieldProps<IRangeSelectSchema>) => {
 	// EVENT HANDLERS
 	// =============================================================================
 	const handleChange = (option: InputRangeProp<IRangeSelectOption>): void => {
-		if (option.from) onChange({ target: { value: { from: option.from.value, to: toStateValue } } });
+		if (option.from == undefined && option.to == undefined)
+			onChange({ target: { value: { from: undefined, to: undefined } } });
+		if (option.from) onChange({ target: { value: { from: option.from.value, to: undefined } } });
 		if (option.to) onChange({ target: { value: { to: option.to.value, from: fromStateValue } } });
 	};
 
+	const handleWrapperBlur = () => {
+		if (!value.from || !value.to) {
+			setToStateValue(undefined);
+			setFromStateValue(undefined);
```

Comment:
can fire `onChange` instead to clear the value in react-hook-form, this ensures no values are submitted if user entered 1 value only

this also solves flickering issue when user tries to select new values again after clearing

Reply:
done